src/Controls/src/Core/PointerGestureRecognizer.cs

```diff
@@ -154,5 +155,36 @@ internal void SendPointerMoved(View sender, Func<IElement?, Point?>? getPosition
 			EventHandler<PointerEventArgs>? handler = PointerMoved;
 			handler?.Invoke(sender, new PointerEventArgs(getPosition));
 		}
+
+		internal static void SetupForPointerOverVSM(
+			VisualElement element,
+			Action<bool> updatePointerState,
+			ref PointerGestureRecognizer? recognizer)
+		{
+			bool hasPointerOverVSM =
+				element.FindFirstVisualStateByName(VisualStateManager.CommonStates.PointerOver) != null;
+
+			if (hasPointerOverVSM && recognizer == null)
+			{
+				recognizer = new PointerGestureRecognizer();
+
+				recognizer.PointerEntered += (s, e) =>
+				{
+					updatePointerState.Invoke(true);
+				};
+
+				recognizer.PointerExited += (s, e) =>
+				{
+					updatePointerState.Invoke(false);
+				};
+
+				((IGestureController)element).CompositeGestureRecognizers.Add(recognizer);
+			}
+			else if (!hasPointerOverVSM && recognizer != null)
+			{
+				((IGestureController)element).CompositeGestureRecognizers.Remove(recognizer);
```

Comment:
What happens if there's no VisualState, but the user is doing something else in response to the pointer entering the control?

Reply:
This is a pointer gesture that's hidden from the user via `IGestureController`. `CompositeGestureRecognizers` isn't something that's exposed to the user, so if the user adds a `PointerGestureRecognizer` it will still remain. This is only adding and removing the `PGR` that we are adding in response to the existence of a `PointerOver` VSM state.